.github/workflows/try_labels.yml

```diff
@@ -0,0 +1,172 @@
+                  repo: context.repo.repo,
+                  body
+                })
+            }
+
+            let try_string = "${{ inputs.try_string }}" || "";
+
+            if (context.eventName == 'pull_request_target') {
+              for (const label of context.payload.pull_request.labels) {
+                if (!label.name.startsWith("T-")) {
+                  continue;
+                }
+
+                // Try to remove the label. If that fails, it's likely that another
+                // workflow has already processed it or a user has removed it.
+                try {
+                  await github.rest.issues.removeLabel({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    issue_number: context.issue.number,
+                    name: label.name,
+                  });
+                } catch (exception) {
+                  console.log("Assuming '" + label.name + "' is already removed: " + exception);
+                  continue;
+                }
+
+                console.log("Found label: " + label.name);
+                try_string += " " + label.name.slice(2); // "T-" removed
+              }
+            }
+
+            console.log(try_string);
+
+            // empty try string
+            if (!try_string) {
+              return "";
+            }
+
+            let username = context.payload.sender.login;
+            let result = await github.rest.repos.getCollaboratorPermissionLevel({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              username
+            });
+            if (!result.data.user.permissions.push) {
+              makeComment('🔒 User @' + username + ' does not have permission to trigger try jobs.');
+              return "";
+            }
+
+            const url = context.serverUrl +
+              "/" + context.repo.owner +
+              "/" + context.repo.repo +
+              "/actions/runs/" + context.runId;
+            const formattedURL = "[#" + context.runId + "](" + url + ")";
+            makeComment("🔨 Triggering try run (" + formattedURL + ")");
+
+            return try_string;
+      - uses: actions/checkout@v3
+        if: github.event_name != 'issue_comment' && github.event_name != 'pull_request_target'
+        with:
+          fetch-depth: 2
+          sparse-checkout: |
+            etc/try_parser.py
+            third_party/ply
+          sparse-checkout-cone-mode: false
+      # This is necessary to checkout the pull request if this run was triggered
+      # via an `issue_comment` action on a pull request.
+      - uses: actions/checkout@v3
+        if: github.event_name == 'issue_comment' || github.event_name == 'pull_request_target'
+        with:
+          ref: refs/pull/${{ github.event.issue.number || github.event.number }}/head
+          fetch-depth: 2
+          sparse-checkout: |
+            etc/try_parser.py
+            third_party/ply
```

Comment:
I don't think ply is necessary any longer.

Reply:
You are right.